odoo/models.py

```diff
@@ -2882,11 +2887,17 @@ def _setup_base(self):
             # avoid clashes with inheritance between different models
             for name in cls._fields:
                 delattr(cls, name)
-            cls._fields = OrderedDict()
-            for name, field in sorted(getmembers(cls, Field.__instancecheck__), key=lambda f: f[1]._sequence):
-                # do not retrieve magic, custom and inherited fields
-                if not any(field.args.get(k) for k in ('automatic', 'manual', 'inherited')):
-                    self._add_field(name, field.new())
+            cls._fields.clear()
+
+            # collect the definitions of each field (base definition + overrides)
+            definitions = defaultdict(list)
+            for klass in reversed(cls.mro()):
+                if getattr(klass, 'pool', None) is None and isinstance(klass, MetaModel):
```

Comment:
What about setting explicitly `pool = None` on `BaseModel`?
On definition classes `klass.pool` is `None`, while on registry classes `klass.pool` is the class' registry.
For the record, in ancient times, the registry was called the "pool", because it is a... collection of... objects :laughing: